The warning for allowImportingTsExtensions is inconsistent with rewriteRelativeImportExtensions

### 🔎 Search Terms

- error TS5096
- allowImportingTsExtensions
- rewriteRelativeImportExtensions
- tsconfig
- diagnostics
- intellisense

### 🕗 Version & Regression Information

- This changed between versions N/A and 5.7.0-beta
- This changed in commit or PR https://github.com/microsoft/TypeScript/pull/59767
- This is the behavior in every version I tried, and I reviewed the FAQ for entries for `allowImportingTsExtensions` and `rewriteRelativeImportExtensions`
- I was unable to test this on prior versions because it was only introduced in 5.7


### ⏯ Playground Link

_No response_

### 💻 Code

tsconfig.json with diagnostic error:

```json
{
  "compilerOptions": {
    "module": "Node16",
    "moduleResolution": "Node16",
    "allowImportingTsExtensions": true,
  }
}
```

### 🙁 Actual behavior

When specifying `allowImportingTsExtensions` as above, I get an error, as intended:

```
tsconfig.json:5:35 - error TS5096: Option 'allowImportingTsExtensions' can only be used when either 'noEmit' or 'emitDeclarationOnly' is set.
``` 

However, if I add `"rewriteRelativeImportExtensions": true` to the same tsconfig, the error goes away, which is contrary to the error message.

The _behavior_ seems correct, but the error message seems incomplete or inconsistent.

---

At the same time, intellisense for `allowImportingTsExtensions` in VSCode reports:

> Allow imports to include TypeScript file extensions. Requires '--moduleResolution bundler' and either '--noEmit' or '--emitDeclarationOnly' to be set.

This is only mentioned as a requirement in intellisense, but not in the error message.

### 🙂 Expected behavior

I would expect the message to reference `rewriteRelativeImportExtensions` as one of the conditions that allow for `allowImportingTsExtensions`. 

I am not sure whether `--moduleResolution bundler` is required, but at least unifying the intellisense and diagnostic would be good. 

The combination of factors seems to be "`noEmit` or `emitDeclarationOnly` or `rewriteRelativeImportExtensions`", but I am not sure what the exact new diagnostic message should be 😅

(I don't mean to be nitpicky! I tend to learn best from intellisense and the diagnostics errors, so I was on the lookout for them when trying out the `5.7.0-beta` and `rewriteRelativeImportExtensions` 😌)

### Additional information about the issue

_No response_

--- a/src/compiler/diagnosticMessages.json
+++ b/src/compiler/diagnosticMessages.json
@@ -4669,7 +4669,7 @@
         "category": "Error",
         "code": 5095
     },
-    "Option 'allowImportingTsExtensions' can only be used when either 'noEmit' or 'emitDeclarationOnly' is set.": {
+    "Option 'allowImportingTsExtensions' can only be used when one of 'noEmit', 'emitDeclarationOnly', or 'rewriteRelativeImportExtensions' is set.": {
         "category": "Error",
         "code": 5096
     },

--- a/src/compiler/program.ts
+++ b/src/compiler/program.ts
@@ -4318,7 +4318,7 @@ export function createProgram(_rootNamesOrOptions: readonly string[] | CreatePro
         }
 
         if (options.allowImportingTsExtensions && !(options.noEmit || options.emitDeclarationOnly || options.rewriteRelativeImportExtensions)) {
-            createOptionValueDiagnostic("allowImportingTsExtensions", Diagnostics.Option_allowImportingTsExtensions_can_only_be_used_when_either_noEmit_or_emitDeclarationOnly_is_set);
+            createOptionValueDiagnostic("allowImportingTsExtensions", Diagnostics.Option_allowImportingTsExtensions_can_only_be_used_when_one_of_noEmit_emitDeclarationOnly_or_rewriteRelativeImportExtensions_is_set);
         }
 
         const moduleResolution = getEmitModuleResolutionKind(options);

--- a/tests/baselines/reference/bundlerImportTsExtensions(allowimportingtsextensions=true,noemit=false).errors.txt
+++ b/tests/baselines/reference/bundlerImportTsExtensions(allowimportingtsextensions=true,noemit=false).errors.txt
@@ -1,6 +1,6 @@
 error TS5056: Cannot write file 'out/b.js' because it would be overwritten by multiple input files.
 error TS5056: Cannot write file 'out/c.js' because it would be overwritten by multiple input files.
-error TS5096: Option 'allowImportingTsExtensions' can only be used when either 'noEmit' or 'emitDeclarationOnly' is set.
+error TS5096: Option 'allowImportingTsExtensions' can only be used when one of 'noEmit', 'emitDeclarationOnly', or 'rewriteRelativeImportExtensions' is set.
 error TS6054: File '/project/e.txt' has an unsupported extension. The only supported extensions are '.ts', '.tsx', '.d.ts', '.js', '.jsx', '.cts', '.d.cts', '.cjs', '.mts', '.d.mts', '.mjs'.
   The file is in the program because:
     Root file specified for compilation
@@ -11,7 +11,7 @@ error TS6054: File '/project/e.txt' has an unsupported extension. The only suppo
 
 !!! error TS5056: Cannot write file 'out/b.js' because it would be overwritten by multiple input files.
 !!! error TS5056: Cannot write file 'out/c.js' because it would be overwritten by multiple input files.
-!!! error TS5096: Option 'allowImportingTsExtensions' can only be used when either 'noEmit' or 'emitDeclarationOnly' is set.
+!!! error TS5096: Option 'allowImportingTsExtensions' can only be used when one of 'noEmit', 'emitDeclarationOnly', or 'rewriteRelativeImportExtensions' is set.
 !!! error TS6054: File '/project/e.txt' has an unsupported extension. The only supported extensions are '.ts', '.tsx', '.d.ts', '.js', '.jsx', '.cts', '.d.cts', '.cjs', '.mts', '.d.mts', '.mjs'.
 !!! error TS6054:   The file is in the program because:
 !!! error TS6054:     Root file specified for compilation

